### ISSUE TITLE
Counting FLOPS

### ISSUE THREAD

**Issue by jahongir7174** [ultralytics/yolov5#2210]
Thank you for sharing the code.
I have a question about the calculation of FLOPS. I checked thop.profile(model, inputs, verbose=True).
But it shows some warning that there are no hooks to count flops for some layers and it is counting as zero macs

![flops](https://user-images.githubusercontent.com/71679734/107855436-ac83c080-6e65-11eb-9e22-e8e9c466319d.PNG)
.

**Comment by glenn-jocher** [comment]
@jahongir7174 that's a good question. I would raise this issue directly on the thop rep (https://github.com/Lyken17/pytorch-OpCounter) regarding the missing official torch nn SiLU layer.

For the custom layers I would not expect them to be included, but they are composed of smaller layers down to the basic nn.Conv2d() layer, which are being included. So for example the Focus() layer itself has no operations other than a reshaping (zero FLOPS), but it has a Conv() layer inside it, which in turn has a Conv2d() layer inside, which is being counted by thop. You can verify this by by profiling individual YOLOv5 layers using our profiling tool (shown in the Colab appendix) and observing their FLOPS:

```python
# Profile
from utils.torch_utils import profile 

m1 = lambda x: x * torch.sigmoid(x)
m2 = torch.nn.SiLU()
profile(x=torch.randn(16, 3, 640, 640), ops=[m1, m2], n=100)
```

**Comment by jahongir7174** [comment]
thank you for your explanation.

**Comment by jahongir7174** [comment]
I tried to make a FLOPS counter for the PyTorch module by using caffe2, I tested mobilenetv2, and it seems it is working properly

https://github.com/jahongir7174/FLOPS-pt

**Comment by glenn-jocher** [comment]
@jahongir7174 ah ok. What values does your FLOPS counter get for the YOLOv5 models at 640?

**Comment by AnjaliBansal2882** [comment]
I would like to ask if there is some way to count the FLOPs in YOLO v8n model using Ultralytics ?

**Comment by pderrenger** [comment]
Yes—Ultralytics YOLO reports GFLOPs for YOLOv8n out of the box. In Python:
from ultralytics import YOLO
model = YOLO('yolov8n.pt')
model.info(verbose=True)  # prints Params and GFLOPs (default imgsz=640)

Any yolo CLI run with model=yolov8n.pt also prints the same summary; adjust imgsz to your target. For context on the metric, see FLOPs: Machine Learning Model Computational Complexity.
